Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/EditorModeOutline.azsl

```diff
@@ -16,44 +16,64 @@ partial ShaderResourceGroup PassSrg : SRG_PerPass_WithFallback
 
     //! Outline line color.
     float4 m_lineColor;
+
+    //! Outline style to use ()
+    uint m_outlineStyle;
 }
 
 //! Applies an outline effect to the entity mask of the specified line thickness.
-//! @note There are known issues with pimples due to z-fighting of coincident surfaces in the entity mask that will be addressed it LYN-10304.
-float Outliner(const float2 uv, const float lineThickness)
+//! @returns The outlines of both channels of the entity mask.
+float2 Outliner(const float2 uv, const float lineThickness)
 {
-    const float2 maskSize = GetImageSize(PassSrg::m_entityMask);
-    const float2 unormalizedUv = float2(uv.x * maskSize.x, uv.y * maskSize.y);
-    const float2 maskSizeReciprocal = float2(1., 1.0) / maskSize;
-    float outline = 0.;
+    const float2 unormalizedUv = float2(uv.x * PassSrg::m_maskDimensions.x, uv.y * PassSrg::m_maskDimensions.y);
+    float2 outlines = float2(0., 0.);
     
     // Outliner kernal
     for(float u = -lineThickness; u <= lineThickness; u+=1.)
     {
         for(float v = -lineThickness; v <= lineThickness; v+=1.)
         {
-            outline += PassSrg::m_entityMask.Sample(PassSrg::PointSampler, (unormalizedUv + float2(u,v)) * maskSizeReciprocal).r;
+            outlines += PassSrg::m_entityMask.Sample(PassSrg::PointSampler, (unormalizedUv + float2(u,v)) * PassSrg::m_maskDimensions.zw).rg;
         }
     }
     
-    // Only apply the outliner effect to fragments not part of S
-    return min(1.0, (outline / (lineThickness * lineThickness)));
+    return min(float2(1.0, 1.0), (outlines / (lineThickness * lineThickness)));
 }
 
 PSOutput MainPS(VSOutput IN)
 {
     PSOutput OUT;
 
     const float3 inColor = PassSrg::m_framebuffer.Sample(PassSrg::LinearSampler, IN.m_texCoord).rgb;
-    const float mask = PassSrg::m_entityMask.Sample(PassSrg::PointSampler, IN.m_texCoord).r;
+    const float2 mask = PassSrg::m_entityMask.Sample(PassSrg::PointSampler, IN.m_texCoord);
+
+    // Only attempt to outline fragments that lie outside of the entity mask
+    if((mask.r * mask.g) > 0.0)
```

Comment:
nit: Missing space after `if` (repeated below) unless this is a different style in shader land?

Reply:
Hmmm there seems to be a 2:1 ratio of `if ()` to `if()` in the shader code base. Mine are consistent accross my shaders so I'd be hesitant to change it right now, but it seems that a linting of the shader code base could be something for the future 👍